experimental/Geometry/Schemes.jl

```diff
@@ -0,0 +1,1244 @@
+    error( "the number of glueings does not coincide with the number of patches so far." )
+  end
+  for i in n
+    if first_patch( glueings[i] ) != get_patches(C)[i] 
+      error( "Domain of the glueing does not coincide with the patch." )
+    end
+    if second_patch( glueings[i] ) != X
+      error( "Codomain of the glueing does not coincide with the new patch." )
+    end
+  end
+  C.glueings = vcat( hcat( C.glueings, glueings ), Array{Nothing,2}(nothing,1,n+1) )
+  return C
+end
+
+#####################################################################
+# intersection of two patches X and Y in a given covering.
+# Returns the intersection as an affine scheme 
+# together with the two open inclusions.
+function intersect( X::AffineScheme, Y::AffineScheme, C::Covering )
+  if !(X in get_patches(C)) 
+    error( "X is not a patch in the covering C!" )
+  end
+  if !(Y in get_patches(C)) 
+    error( "Y is not a patch in the covering C!" )
+  end
+  if X == Y 
+    return ( X, get_identity(X), get_identity(X) )
+  end
+  i = indexin( X, get_patches(C) )[1]
+  j = indexin( Y, get_patches(C) )[1]
+  # Keep in mind that the glueing is realized 
+  # in a directed way. Thus, we need to make a distinction.
+  if i < j
+    G = get_glueings(C)[i,j]
+    return (overlap(G),first_inclusion(G),second_inclusion(G))
+  else
+    G = get_glueings(C)[j,i]
+    return (overlap(G),second_inclusion(G),first_inclusion(G))
+  end
+end
+
+###############################################################
+# Restriction of a morphism to open inclusions:
+#    f
+#  X → Y
+# i↑   ↑j
+#  U → V
+#    f'
+# When f, i, and j are given where X and Y are affine schemes 
+# and i and j are inclusions of principal open subsets with 
+# f(U) ⊂  V, we would like to explicitly infer the restriction 
+# f' of f to U.
+function restrict( f::AffSchMorphism, U::SpecPrincipalOpen, V::SpecPrincipalOpen )
+  # TODO: implement this.
+end
+
+##############################################################
+# Take the union of two affine patches in a given covering.
+function union( X::AffineScheme, Y::AffineScheme, C::Covering )
+  # TODO: This is not functional yet. We first need a struct 
+  # for morphisms of CoveredScheme.
+end
+
+#######################################################################
+#
+# Refinement of a covering C by another covering D.
+# Let I and J be the index sets for the patches of the coverings 
+# C and D, respectively. Then a refinement consists of 
+# a function 
+#   n : J → I 
+# such that for every patch Y_j in D there exists a patch X_i in C with 
+# i = n(j) such that X_i contains Y_j. 
+# The datum of the refinement stored here is then the respective collection 
+# of morphisms of affine schemes given by these containments. 
+#
+mutable struct Refinement
+  original_cover::Covering
+  refined_cover::Covering
+  index_map::Vector{Int}
+  inclusions::Vector{AffSchMorphism}
+end
+
+function get_original_cover( ρ::Refinement )
```

Comment:
These names (`get_foo`) are not idiomatic (e.g. we don't have `get_parent`, but rather `parent`).

Reply:
The drawback is that one has to remember the order of the arguments.

Reply:
I don't understand, there's only one argument here? And how does `get_foo` vs. `foo` help to indicate the order of arguments?

Reply:
It was about the 
""" experimental/Geometry/Schemes.jl
    x = new{S,Td,Ud,Tc,Uc}()
    x.domain = domain
    x.codomain = codomain
    x.pullback = pullback
@fingolfin fingolfin 1 hour ago Member

It's more idiomatic, and more concise.""" comment. There was no answer button there. Sorry.